.changeset/five-dolls-smash.md

```diff
@@ -0,0 +1,20 @@
+---
+'@astrojs/svelte': major
+---
+
+Adds support for Svelte 5. Svelte 3 and 4 are no longer supported.
```

Comment:
Just checking on this statement, because there seems to be a note in the docs PR about a way to add legacy 3 and 4 support.

Will Astro 5 require v5 of this integration? Or is this entirely independent? I'm guessing that if Astro 5 requires v5, that whenever *that* change is made to `astro` itself, there will be a separate changeset mentioning that?

But here, it might be worth mentioning if this version is incompatible with versions of Astro pre-Astro 5. (Can someone upgrade `@astrojs/svelte` to v5 in a v4 project? It's possible someone might want/try to do that if they're not ready to upgrade to *Astro 5*, but do want to just upgrade to use *Svelte 5*.)

Reply:
Currently this PR is only to bump a major version of `@astrojs/svelte` (merges into `main`). It's not related to Astro 5. It's only to support Svelte 5 for now. Later if Astro 5 requires another breaking change of `@astrojs/svelte`, we may bump another major for it, but so far it's not needed.

Currently `@astrojs/svelte` is v5, which supports Svelte v3 and v4. This PR bumps `@astrojs/svelte` to v6, which supports Svelte v5. So for Svelte v3 and v4 support, they need to use `@astrojs/svelte` v5.

Since Astro 5 doesn't require any breaking changes for `@astrojs/svelte` yet, in the future we can extend the `astro` peer dependency range for both `@astrojs/svelte` v5 and v6 if needed so that users can choose either one.

Hope that clears up. Unfortunate numbers all around 😄

Reply:
hahaha... oh my. Yes, the numbers are all quite unfortunate, but gotcha now, thanks! I don't know why I had it in my head this was to `next` and not `main`. :sweat_smile: